cmd/prometheus/main.go

```diff
@@ -558,6 +558,12 @@ func main() {
 	}
 	defer closer.Close()
 
+	listener, err := webHandler.Listener()
```

Comment:
I'd rather we not put startup logic outside of the run.Group paradigm we currently have. If you want this to block other things you should do it with a channel, as we do with the tsdb itself starting up.

I suspect that this approach could also lead to the listener leaking for some other users of this code.

Reply:
I think this change is fine. I don't know anyone using main as you can't import the package.

Also we do this for checking the config file, too.

Reply:
Hmm, okay.